### DISCUSSION TITLE
Trigger the ChangeContainerOperation

### DISCUSSION THREAD

**Discussion by SchmidtMarc** [eclipse-glsp/glsp#731] [Q&A] [answered]
Hello,

I'm currently implementing a GLSP-based editor for hierarchical state machines. With the help of the discussion [692](https://github.com/eclipse-glsp/glsp/discussions/692), I managed to nest nodes using compartments. Now I want to implement the `ChangeContainerOperation` but I couldn't find out how to trigger this action. I expected dragging and dropping a contained node to another container would work, but it doesn't.
It seems like there is no implementation of the `ChangeContainerOperation` in the workflow example.
The questions:
* What triggers this action?
* Do we have to implement something on the client side?

Thanks for your help
Marc Schmidt

**Comment by CamilleLetavernier** [comment, marked as answer]
Hi Marc,

At the moment, this isn't implemented in the example project, nor in the base framework. To support this, one needs to update the `ChangeBoundsTool` in `change-bounds-tool.ts`, to trigger a ChangeContainerOperation when the target node is different from the current parent (And when the dragged node supports reparenting; based on the current ShapeTypeHints configuration)

The corresponding handler must also be added on the server.

There may also be issues with Feedback/Z-index during the move/reparent operation: https://github.com/eclipse-glsp/glsp/issues/501

**Reply by SchmidtMarc** [reply]
Hello Camille,

thanks for your reply and the rough overview of what needs to be done for this feature. When I find time and understand the client code better, I will implement it and create a pull request.
